ndb/MIGRATION_NOTES.md

```diff
@@ -123,6 +123,30 @@ The primary differences come from:
   is also a user-defined property named `key`. For an example, see the
   class docstring for `Model`.
 
+## Bare Metal
+
+One of the largest classes of differences comes from the use of the current
+Datastore API, rather than the legacy App Engine Datastore. In general, for
+users coding to the public interface, this won't be an issue, but users relying
+on pieces of the ostensibly private API that are exposed to the bare metal of
+the original datastore implementation will have to rewrite those pieces.
+Specifically, any function or method that dealt directly with protocol buffers will
+no longer work. Datastore, itself, has moved to gRPC and protobuffers
+themselves are significantly different from legacy Datastore.  Additionally,
+this version of NDB makes less direct use of protobuffers and relies more
+heavily on Datastore for parsing and generating protobuffers, which is a
+significant internal refactoring.
+
+- `ModelAdapter` is no longer used. In legacy NDB, this was passed to Datastore
+  so that calls to Datastore RPCs could yield NDB entities directly from
+  Datastore RPC calls. AFAIK, Datastore no longer accepts an adapter for
+  adpating entities. At any rate, we no longer do it that way.
+- `Property._db_get_value` is no longer used. It worked directly with Datastore
+  protobuffers, work which is now delegated to Datastore.
```

Comment:
@dhermes You mention above dropping use of "protobuffers"--how should I refer to the `p` argument of the new deprecated `Property._db_get_value`?

Reply:
If it's deprecated, do we need to document it?

Reply:
But also, you can just say "protocol buffer" every time you were using "protobuffer".

Reply:
I think it's similar status to a lot of stuff in this file: probably not an issue but worth mentioning in case someone was overriding `Property._db_get_value` for some ill-advised reason. I get the impression that best practices for NDB evolved over time. There's an element of archaeology to this work.

Reply:
:+1:

Reply:
ab2f965